components/channel_view/channel_view.jsx

```diff
@@ -70,6 +71,21 @@ export default class ChannelView extends React.PureComponent {
         return this.refs.channelView;
     }
 
+    componentDidUpdate(prevProps) {
+        if (prevProps.channelId !== this.props.channelId) {
+            mark('ChannelView#componentDidUpdate');
+
+            measure('SidebarChannelLink#click', 'ChannelView#componentDidUpdate');
+            measure('TeamLink#click', 'ChannelView#componentDidUpdate');
+
+            clearMarks([
+                'SidebarChannelLink#click',
+                'ChannelView#componentDidUpdate',
+                'TeamLink#click'
+            ]);
+        }
+    }
+
```

Comment:
`measure` returns the duration and the measurement name so we can do this to send the data to Segment:
```js
const [duration, name] = measure('TeamLink#click', 'ChannelView#componentDidUpdate');
trackEvent('ui-performance', name, { duration });
```

Reply:
If you want to add this in, we do run pre-release in developer mode and could get some data out of it

Reply:
Cool I'm going to add it in; expect another commit.